app/ui/src/services/api.ts

```diff
@@ -138,3 +139,62 @@ export async function streamApi(
 
   startStream()
 }
+
+export async function getAgents() {
+  try {
+    const response = await callApi(agentsPath);
+    if (!response.ok) {
+      const errorData = await response.json();
+      throw new Error(errorData.detail || 'Failed to fetch agents.');
+    }
+    return response.json();
+  } catch (error) {
+    console.error('Error fetching agents:', error);
+    throw error;
+  }
+}
+
+export async function addAgent(agent: { name: string; guideline_prompt: string; type: string; }) {
+  try {
+    const response = await callApi(agentsPath, 'POST', agent);
+    if (response.status === 201) {
+      return response.json();
+    }
+    const errorData = await response.json();
+    throw new Error(errorData.detail?.description || 'Failed to add agent.');
+  } catch (error) {
+    console.error('Error adding agent:', error);
+    throw error;
+  }
+}
+
+export async function updateAgent(
+  agent_id: string,
```

Comment:
No snake casing in JS ;)
```suggestion
  agentId: string,
```

Reply:
We could look at this ... https://medium.com/analytics-vidhya/camel-case-models-with-fast-api-and-pydantic-5a8acb6c0eee 🤔